opentelemetry-etw-logs/src/logs/exporter/options.rs

```diff
@@ -0,0 +1,256 @@
+use std::borrow::Cow;
+
+use opentelemetry_sdk::logs::SdkLogRecord;
+
+use crate::logs::processor::{ProcessorBuildError, ProcessorBuildErrorKind};
+
+type BoxedEventNameCallback = Box<dyn EventNameCallback>;
+
+#[derive(Debug)]
+pub(crate) struct Options {
+    provider_name: Cow<'static, str>,
+    event_name_callback: Option<BoxedEventNameCallback>,
+}
+
+impl Options {
+    pub(crate) fn builder(provider_name: impl Into<Cow<'static, str>>) -> OptionsBuilder {
+        OptionsBuilder::new(provider_name)
+    }
+
+    /// Returns the provider name that will be used for the ETW provider.
+    pub(crate) fn provider_name(&self) -> &str {
+        &self.provider_name
+    }
+
+    /// Returns the default event name that will be used for the ETW events.
+    pub(crate) fn default_event_name(&self) -> &str {
+        "Log"
+    }
+
+    /// Returns the event name to be used for the ETW event given the log record contents and the options.
+    pub(crate) fn get_etw_event_name<'a>(
+        &'a self,
+        log_record: &'a opentelemetry_sdk::logs::SdkLogRecord,
+    ) -> &'a str {
+        if let Some(callback) = &self.event_name_callback {
+            let result = callback(log_record);
+            if validate_etw_event_name(result).is_ok() {
+                return result;
+            }
+        }
+        self.default_event_name()
+    }
+}
+
+trait EventNameCallback: Fn(&SdkLogRecord) -> &str + Send + Sync + 'static {}
+
+impl<F> EventNameCallback for F where F: Fn(&SdkLogRecord) -> &str + Send + Sync + 'static {}
+
+impl std::fmt::Debug for dyn EventNameCallback {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("ETW event name callback")
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct OptionsBuilder {
```

Comment:
its easier to review, if builder is handled completed inside the ProcessorBuilder itself.